### PR TITLE
Drupal 9 updates

### PR DESCRIPTION
**Pull request by ben-hamelin**
## Summary

Create an updated project starting point for Drupal 9. Includes composer updates, documentation, move to /docroot/ and some other misc cleanup

## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes/no
| Did you apply meaningful labels to the pull request? | yes/no
| Did you perform a self review first? | yes/no
| Documentation reflects changes? | yes/no
| `CHANGELOG` reflects changes? | yes/no
| Unit/Functional tests reflect changes? | yes/no
| Did you perform browser testing? | yes/no
| Risk level | Medium
| Relevant links | JIRA issue, bug reports, security alerts, etc.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -11,15 +11,15 @@ const fileArgs = require('yargs').argv;
 // Directories to search SCSS files to compile. By default, node-sass does not
 // compile files that begin with _.
 const scssFilePaths = [
-  "web/modules/custom/**/*.scss",
-  "web/themes/custom/**/*.scss",
+  "docrootmodules/custom/**/*.scss",
```
Comment (raubin) [inline gulpfile.js:14]: did you mean to have `docroot/modules/` and `docroot/themes/` instead?

File: web/sites/default/settings.php [hunk 2]
```diff
@@ -1,51 +0,0 @@
-<?php
```
Comment (raubin) [inline web/sites/default/settings.php:1]: With this deleted, is it generated automatically by d9?

File: gulpfile.js (hunk 1 above)
Comment (ben-hamelin) [reply gulpfile.js]: Oops! Thanks

File: web/sites/default/settings.php (hunk 2 above)
Comment (ben-hamelin) [reply web/sites/default/settings.php:1]: Following suite from the last few projects, specifically Acquia hosted, that we've worked on.

File: .gitignore [hunk 3]
```diff
@@ -1,35 +1,36 @@
 # Drupal
-/web/*.*
-/web/core
-/web/libraries
-/web/modules/contrib
-/web/themes/contrib
-/web/profiles/contrib
+/docroot/*.*
```
[the remark this answers is not in the document]
Comment (ben-hamelin) [reply .gitignore:2]: Following suite from the last few projects, specifically Acquia hosted, that we've worked on.

File: web/sites/default/settings.php (hunk 2 above)
Comment (ben-hamelin) [reply web/sites/default/settings.php:1]: The docroot/sites/default/ equivalent is present. It still isn't there in the core download

**Comment by ben-hamelin** [comment]
@raubin made those fixes, thanks. Can you take this for a test drive locally and see how it works for you?

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (raubin) [reply .gitignore:2]: I checked with @edaemon and our CI setup assumes `web`, so let's roll this back and stick with that for now. `docroot` is vendor specific just for Acquia and won't be our preference for other setups.

**Comment by raubin** [comment]
I'll give this a run through locally, but noticed that my version of Lando doesn't support the d9 recipe so I need to upgrade it first.

File: .gitignore [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
-!/web/sites/default
-/web/sites/default/*
-!/web/sites/default/settings.php
+/docroot/sites/*
+!/docroot/sites/default
+/docroot/sites/default/*
+!/docroot/sites/default/settings.php
 
-!/web/.rsync-filter
+!/docroot/.rsync-filter
```
[the remark this answers is not in the document]
Comment (ben-hamelin) [reply .gitignore:14]: removed

File: .editorconfig [hunk 5]
```diff
@@ -1,17 +0,0 @@
-# Drupal editor configuration normalization
```
[the remark this answers is not in the document]
Comment (ben-hamelin) [reply .editorconfig]: Removed from ignore file, committed back

**Comment by ben-hamelin** [comment]
@raubin Addressed comments. I will continue to test d9 locally

File: web/modules/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+/README.txt
```
Comment (raubin) [inline web/modules/.gitignore:1]: missing new line

File: web/sites/default/settings.php [hunk 7]
```diff
@@ -1,51 +1,783 @@
...
+ * caching, JavaScript/CSS compression, re-routing of outgoing emails, and
+ * other things that should not happen on development and testing sites.
+ *
+ * Keep this code block at the end of this file to take full effect.
+ */
+#
+# if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
+#   include $app_root . '/' . $site_path . '/settings.local.php';
+# }
+$databases['default']['default'] = array (
```
Comment (raubin) [inline web/sites/default/settings.php:773]: I'm fine adding in the d9 boilerplate but we should probably keep the dynamic Lando config we had so we're not hardcoding recipe values and only acting on a Lando environment.

**Comment by raubin** [comment]
@ben-hamelin - my d9 install testing went well but flagged a couple things above before we merge this in.

File: web/sites/default/settings.php (hunk 7 above)
Comment (ben-hamelin) [reply web/sites/default/settings.php]: Ok restored those

**Comment by ben-hamelin** [comment]
@raubin Addressed those two notes

**Comment by raubin** [comment]
Cool, thanks @ben-hamelin ! /approved

**Pull request merged**